### DISCUSSION TITLE
esp32 additional setting / switch

### DISCUSSION THREAD

**Discussion by ApoGedon** [GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200#32] [General]
Hallo GyroGearl00se,

I recreated your project on my PE1 Pellet 25 and everything worked great.
In the last 3 weeks I went from setting up HAOS and a few Zigbee Switches to This and
here I lack experience in adding additional parameters / entities to the esp32.

Is it possible to switch the "Heizkreis - Betriebsart"?
![Screenshot 2024-06-11 142836](https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/assets/171260812/bd01b27c-0719-4852-bf36-0542b4ce4be8)

As far I understand is that I have to add code to heizkreis01.yaml

Something like this, but probably wrong.

```
- platform: modbus_controller
    modbus_controller_id: ${modbus_controller_id}
    name: "${name} Heizkreis01 Betriebsart"
    id: ${name}_heizkreis01_betriesart           # <- is this the name that appears as entity in HA?
    address: 48047
    unit_of_measurement: "°C"           # <- not needed?
    value_type: U_WORD
    multiply: 2               # <- also no clue 
    min_value: 0
    max_value: 5

```

Then I need to add something to the froeling.yaml?
Sorry for the lack of knowledge here, if it is easy I would really appreciate the example code.
If it is complex I will research more in how to work with esp32.

Wo kann ich dir ein Bier ausgeben für das super Projekt, (Paypal?)  wirklich toll.
Grüße ApoGedon

**Comment by GyroGearl00se** [comment]
Hi @ApoGedon,
freut mich das dieses Projekt dir gefällt.

Ich bin mir hierüber nicht ganz sicher wie ich das am besten umsetzen könnte. Müsste ich mir bei Gelegenheit mal genauer anschauen und auch testen bevor ich es im code verfügbar mache.
Aber wenn ich mir dir ESPHome doku ansehe, denke ich das ich vermutlich wie folgt ansetzen würde:

select:
  - platform: modbus_controller
    modbus_controller_id: ${modbus_controller_id}
    name: "${name} Heizkreis01 Betriebsart"
    id: ${name}_heizkreis01_betriesart
    address: 8047
    value_type: U_WORD
    optionsmap:
      "Aus": 0
      "Automatik": 1
      "Extraheizen": 2
      "Absenken": 3
      "Dauerabsenken": 4
      "Partybetrieb": 5

Eventuell braucht es noch lambda &oder write_lambda parameter.

Ich werde versuchen mir das in den nächsten Wochen mal anzuschauen sofern ich dazu komme. Solltest du schon eher zu einer Lösung kommen, würde ich mich natürlich über eine Rückmeldung freuen.

Ein Spendenkonto oder ähnliches habe ich nicht :)
Freut mich zu hören das es dir das Wert wäre - das genügt. Danke.

**Reply by ApoGedon** [reply]
Hallo @GyroGearl00se ,

danke für die Antwort! Hab das Beispiel mal in den code reingeworfen, kommen beim 
installieren ein paar fehler :)  
Ich hab beim recherchieren ein Beispiel bei der rs232 - tpc umsetzung gefunden.
Vielleicht hilft das.
https://github.com/smokyflex/pe1-modbus?tab=readme-ov-file#remote-control

Kein Stress ob oder wann du dazu kommst.

Grüße ApoGedon

**Comment by GyroGearl00se** [comment]
Hi @ApoGedon,
Ich habe es mir mal angesehen und nun die Umsetzung hier im Git für Heizkreis 01 & 02 entsprechend eingepflegt.
https://github.com/GyroGearl00se/ESPHome-Froeling-Lambdatronic_3200/commit/b40e574322555554cafc33a60d36c5dd8efc69a4

Gerne mal testen und zurückmelden ob das auch bei dir funktioniert wie man es erwarten würde.

**Reply by ApoGedon** [reply]
Hey @GyroGearl00se es funktionier richtig geil!
Hab alle beide Heizkreise ausprobieren können, gehen beide perfekt zum steuern.
Falls mir ein fehler auffällt geb ich bescheid.

Made my day.
Grüße ApoGedon
PS: Mein Angebot für ein Trinkgeld steht immernoch :)

**Reply by GyroGearl00se** [reply]
Freut mich das es wie erwartet funktioniert 👍 

Danke für das Angebot - Ich bevorzuge es meine Persönlichen Daten nicht öffentlich zu teilen. Und der Aufwand mit BuyMeACoffee oder ähnlichen Spendenkonten ist es vermutlich nicht wert.

Ich schätze jedoch sehr deine Bereitschaft solche öffentlichen Projekte zu unterstützen. Dankeschön
